metrics-operator/converter/slo_converter.md

```diff
@@ -0,0 +1,387 @@
+The converter is part of `metrics-operator` image.
+
+## Usage
+
+The converter will convert a single `slo.yaml` file into single `AnalysisDefintion` resource.
+
+To run the converter, execute the following command:
+
+```shell
+METRICS_OPERATOR_IMAGE=<METRICS_OPERATOR_IMAGE>
+PATH_TO_SLO=<PATH_TO_SLO>
+ANALYSIS_VALUE_TEMPLATE_NAMESPACE=<ANALYSIS_VALUE_TEMPLATE_NAMESPACE>
+DEFINITION_NAME=<DEFINITION_NAME>
+
+docker-run $METRICS_OPERATOR_IMAGE manager --convert-slo=$PATH_TO_SLO --slo-namespace=$ANALYSIS_VALUE_TEMPLATE_NAMESPACE --definition=$DEFINITION_NAME
+```
+
+Please be aware, you need to substitute the placeholders with the following information:
+
+* **PATH_TO_SLO** - path to your `slo.yaml` file
+* **ANALYSIS_VALUE_TEMPLATE_NAMESPACE** - namespace of `AnalysisValueTemplate` which will be referenced in objectives
+* **DEFINITION_NAME** - name of created `AnalysisDefinition`
+
+> **Note**
+
+All the SLOs present in `slo.yaml` file will reference `AnalysisValueTemplate` resources from the namespace defined
+by `ANALYSIS_VALUE_TEMPLATE_NAMESPACE` argument.
+
+## Conversion details
+
+We have multiple use-cases which are and which are not supported.
+There is a need to convert the use-cases that make
+logical sense and are common, but in some cases, where it is problematic and these cases will not be supported.
+
+> **Note** Please be aware, that comparison criteria containing `%` symbol ware not supported and will be ignored.
+
+### Unsupported use-cases
+
+Criteria with 3 and more inputs won't be supported, only the first 2 non-percentage inputs (those not containing `%`
+as we do not support comparison rules) will be taken and converted.
+In the example below, only `<600` and `>400` rules will be converted.
+Rule `>800` will be ignored.
+
+```yaml
+objectives:
+- sli: response_time_p95
+  displayName: "Response Time P95"
+  pass:
+  - criteria
+    - "<600"
+    - ">400"
+    - ">800"
+  weight: 2
+  key_sli: true
+```
+
+Secondly, conversion of multiple criteria elements won't be supported at all.
+Rules containing multiple criteria elements combined with logical OR (see documentation
+[here](https://github.com/keptn/spec/blob/master/service_level_objective.md#objectives))
+can not be converted and will be turned into informative items.
+Informative in this context means that there are no objectives that have to be met and
+therefore the item will not affect the overall score.
+However, their values are still retrieved to provide additional insights for the execution of an Analysis.
+An example of an unsupported SLOs for conversion is the following:
+
+```yaml
+objectives:
+- sli: response_time_p95
+  displayName: "Response Time P95"
+  pass:
+  - criteria:
+    - "<=+10%"
+  - criteria
+    - "<600"
+  weight: 2
+  key_sli: true
+```
+
+Thirdly, a case where `pass` criteria are set, `warn` criteria are set, but pass criteria interval and warn
```

Comment:
instead of secondly, thirdly, I'd would suggest having a proper name for these cases and add a subsection for them. This way, it is easier with a TOC on top of the document to jump directly to the conversion case